sal-service/src/main/java/org/ow2/proactive/sal/service/service/VaultService.java

```diff
@@ -0,0 +1,119 @@
+import org.ow2.proactive.sal.service.service.application.PASchedulerGateway;
+import org.ow2.proactive.scheduler.common.exception.NotConnectedException;
+import org.ow2.proactive.scheduler.common.exception.SchedulerException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import lombok.extern.log4j.Log4j2;
+
+
+@Log4j2
+@Service("VaultService")
+public class VaultService {
+
+    @Autowired
+    private PAGatewayService paGatewayService;
+
+    @Autowired
+    private PASchedulerGateway schedulerGateway;
+
+    @Autowired
+    private ServiceConfiguration serviceConfiguration;
+
+    private Set<String> getKeys() {
+        try {
+            Set<String> keys = schedulerGateway.thirdPartyCredentialsKeySet();
+            return keys;
+        } catch (SchedulerException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public Boolean registerNewSecrets(String sessionId, Map<String, String> secret) throws NotConnectedException {
+        if (!paGatewayService.isConnectionActive(sessionId)) {
+            throw new NotConnectedException();
+        }
+        Validate.notNull(secret, "No secret where received in the request body.");
+        Set<String> keys = getKeys();
+        secret.forEach((key, value) -> {
+            LOGGER.info("registerNewSecrets endpoint is called to add key: \"{}\".", key);
+            if (keys.contains(key)) {
+                LOGGER.warn("A secret with the same key \"{}\" exist! the value will be overwritten!", key);
+            }
+            try {
+                schedulerGateway.putThirdPartyCredential(key, value);
+            } catch (SchedulerException e) {
+                throw new RuntimeException(e);
+            }
+        });
+        return true;
+    }
+
+    public Boolean removeSecret(String sessionId, String key) throws NotConnectedException {
+        if (!paGatewayService.isConnectionActive(sessionId)) {
+            throw new NotConnectedException();
+        }
+        Validate.notNull(key, "No key was received in the request path.");
+        Set<String> existingKeys = getKeys();
+        LOGGER.info("removeSecret endpoint is called to remove key: \"{}\".", key);
+        if (!existingKeys.contains(key)) {
+            LOGGER.error("The key \"{}\" does not exist in the vault!", key);
+            throw new RuntimeException("the key does not exist in the vault!");
+        }
+        try {
+            schedulerGateway.removeThirdPartyCredential(key);
+        } catch (SchedulerException e) {
+            throw new RuntimeException(e);
+        }
+        return true;
+    }
+
+    public Boolean removeAllSecrets(String sessionId) throws NotConnectedException {
+        if (!paGatewayService.isConnectionActive(sessionId)) {
+            throw new NotConnectedException();
+        }
+        LOGGER.info("removeAllSecrets endpoint is called.");
+        Set<String> keys = getKeys();
```

Comment:
I think this is risky since ProActive could have keys stored from the GUI from other users.
What you should do is maybe keep in SAL's DB the names of the stored keys and ONLY delete the ones handled by SAL.

Reply:
done